### ISSUE TITLE
Meta information does not properly update during TransformInverter

### ISSUE THREAD

**Issue by ZiyueXu77** [Project-MONAI/MONAI#2367]
**Describe the bug**
TransformInverter correctly updated the data array, but the meta information does not get properly updated.
1. "image_meta_dict" will be updated along the pre-transforms and will stay the value before TransformInverter.
2. "pred_inverted_meta_dict" which is expected to be created along with the pred_inverted dict will not be created/available.
3. Thus, the final saved result will be a combination of: correct image array (pred_inverted) + wrong meta dict (image_meta_dict) 

**To Reproduce**
With Clara inference MMAR:
1. Add orientationd and spacingd
    {
      "name": "Orientationd",
      "args": {
        "keys": [
          "image"
        ],
        "axcodes": "LPS"
      }
    },
    {
      "name": "Spacingd",
      "args": {
        "keys": ["image"],
        "pixdim": [0.3, 0.3, 1.0],
        "mode": ["bilinear"],
        "align_corners": [true]
      }
    }
2. Modify /opt/monai/monai/handlers/segmentation_saver.py to print out the affine information before the saving 
            print("SegSaver affine: ", meta_data['affine'])
            print("SegSaver original_affine: ", meta_data['original_affine'].detach().cpu())
            print("engine output: ", engine_output.shape)
            self._saver.save_batch = True
            self._saver(engine_output, meta_data)
3. Modify /opt/monai/monai/transforms/spatial/dictionary.py to print out the affine information during each transform
4. Modify /opt/monai/monai/handlers/transform_inverter.py to track the "# save the inverted meta dict into state.batch" behavior.
5.  You will notice: 1) the "# save the inverted meta dict into state.batch" will do nothing because the key does not exist in engine.state.batch, 2) transforms correctly made transformation / inverse transformations to both data array and metadata, 3) segsaver affine is the wrong one. 

**Expected behavior**
The expected behavior is that the "pred_inverted" which is created by transforminverted, will have its own meta_dict: first copied from image_meta_dict (or pred_meta_dict, which is not there either), and will be correctly updated until the segmentationsaver used it.

**Current solution**
Currently, we can bypass this issue by making a copy of the original meta_dict after LoadImaged as:
     {
          "name": "CopyItemsd",
          "args": {
              "keys": [
                  "image_meta_dict"
              ],
              "times": 1,
              "names": [
                  "pred_expected_meta_dict"
              ]
          }
      }
Then in the end use this "pred_expected_meta_dict" for segmentationsaver:
    {
      "name": "SegmentationSaver",
      "args": {
        "resample": false,
        "output_dir": "{MMAR_EVAL_OUTPUT_PATH}",
        "batch_transform": "lambda x: x['pred_expected_meta_dict']",
        "output_transform": "lambda x: __import__('monai').transforms.AsDiscrete(argmax=True, to_onehot=False, n_classes=2)(x['pred_inverted'][0][None].detach().cpu())"
      }
    }
But this in fact discarded all the meta information updates during the entire pipeline, so is simply bypassing the issue instead of solving it.

**Screenshots**

**Comment by Nic-Ma** [comment]
Hi @ZiyueXu77 ,

Thanks very much for your detailed issue report!
To optimize the API usage, we already simplified the `Invertd` transform args and make it `in-place` operation as other transforms.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/post/dictionary.py#L417

Thanks.
